### PR TITLE
Add release step to CI workflow

### PR DESCRIPTION
**Pull request by rfomin**
When a new tag is pushed to the repository, the release will be automatically generated using RELEASE_NOTES.md and artifacts.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -150,6 +150,6 @@ jobs:
         uses: ncipollo/release-action@v1
         with:
           name: ${{ env.PACKAGE_STRING }}
-          bodyFile: CHANGELOG.md
+          bodyFile: RELEASE_NOTES.md
```
Comment (turol) [inline .github/workflows/main.yml:153]: Why?

Comment (rfomin) [reply .github/workflows/main.yml:153]: There is no `CHANGELOG.md` file in the Chocolate Doom repository, but there is one in Crispy Doom. I renamed it to `RELEASE_NOTES.md` for clarity. Perhaps I should squash these commits.

Comment (turol) [reply .github/workflows/main.yml:153]: There is a `ChangeLog` (without `.md`). Should either use that directly or convert it to markdown.

Comment (rfomin) [reply .github/workflows/main.yml:153]: This is the content of `ChangeLog`:
```
Chocolate Doom's version history is stored in Git. For a full
change log, clone the repository on Github:

    https://github.com/chocolate-doom/chocolate-doom
```

We have `NEWS.md`, but it's too long and too verbose. I think the point of `RELEASE_NOTES.md` is to have the most important news for regular users to read.

Comment (turol) [reply .github/workflows/main.yml:153]: Is there a way to automatically cut `NEWS.md` so it's reasonable?

Comment (rfomin) [reply .github/workflows/main.yml:153]: Perhaps, but we can just copy and paste it into `RELEASE_NOTES.md` and maybe trim it down a bit.

**Comment by rfomin** [comment]
Example (our package version is still 3.0.0): https://github.com/rfomin/chocolate-doom/releases/tag/chocolate-doom-4.0.0

**Review by fabiangreffrath — APPROVED** [review]
I'm all for it! Let's dump Windows XP support for the sake of all new SDL libraries and fluidsynth support out of the box.

**Review by turol — APPROVED** [review]

**Comment by fabiangreffrath** [comment]
Alright, guys, I just got approval from @fragglet by PM that he is fine with us dropping chocpkg and using the CI artifacts as release binaries. One question remains, though, and this is if it's possible to upload the macOS artifacts as well?

**Comment by fabiangreffrath** [comment]
Then we're good to go!

**Pull request merged**